src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/bucketize/bucketize_kernel_ref.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "bucketize_kernel_ref.hpp"
+
+#include "kernel_selector_utils.h"
+
+namespace kernel_selector {
+
+namespace {
+
+CommonDispatchData SetDefault(const bucketize_params& kernel_params) {
+    CommonDispatchData dispatch_data;
+    const auto in_layout = kernel_params.inputs.front().GetLayout();
+    const auto& output = kernel_params.outputs.front();
+    const auto out_layout = output.GetLayout();
+    std::vector<std::vector<Tensor::DataChannelName>> dims_by_gws;
+
+    switch (out_layout) {
```

Comment:
if it's not critical for performance you can use 
 dispatchData.gws = {in.X().v, in.Y().v * in.Z().v * in.W().v, in.Feature().v * in.Batch().v};

and avoid switch (in  less dimension cases Z and W volume would be just 1)

Reply:
Yeah, I can avoid it here, but still, I would need to add more calculations to the kernel and I don't think it would be less in general. And also this switch help to calculate better local work group size by using GetOptimalLocalWorkGroupSizes function (not sure if it works great, but still).

Reply:
you can leave workgroupsize in switch but simplify gws I think

Reply:
But then I need to modify the kernel, right? Not sure whether this change improves or simplifies something, but maybe I just don't understand.